### PR TITLE
`conduct load` and `shazar` now accept stdin input, support new digest format

### PR DESCRIPTION
**Pull request by longshorej**
***Updated 2017-03-08***

`conduct load` and `shazar` have both been updated to stream input from `stdin` and handle a new format where the digest is at the end of the file or stream. Below are manual tests that are also captured in this PR's automated tests:

# Manual Tests

**Loading new format on disk**
```bash
$ tar c conductr-crash-v0 | shazar -o new-format.zip
$ conduct load new-format.zip 
Retrieving bundle..
Retrieving file:///home/longshorej/work/conductr-crash/target/bundle/new-format.zip
Loading bundle to ConductR..
[#################################################] 100%
Bundle 9274f89add3d963ab17ab3df5ce50d1a is installed
Bundle loaded.
Start bundle with: conduct run 9274f89
Unload bundle with: conduct unload 9274f89
Print ConductR info with: conduct info
-> 0
```

**Loading new format via pipes**
```bash
$ tar c conductr-crash-v0 | shazar | conduct load
Retrieving bundle..
Loading bundle from cache typesafe/bundle/-
Bintray credentials loaded from /home/longshorej/.lightbend/commercial.credentials
Loading bundle to ConductR..
[#################################################] 100%
Bundle 9274f89add3d963ab17ab3df5ce50d1a is installed
Bundle loaded.
Start bundle with: conduct run 9274f89
Unload bundle with: conduct unload 9274f89
Print ConductR info with: conduct info
-> 0
```

**Loading old format on disk**
```bash
$ conduct load conductr-crash-v0-644a7fdbf335af40bbe1e00cce3654482d21fc75b6dd3153a6bbb6f19f499129.zip
Retrieving bundle..
Retrieving file:///home/longshorej/work/conductr-crash/target/bundle/conductr-crash-v0-644a7fdbf335af40bbe1e00cce3654482d21fc75b6dd3153a6bbb6f19f499129.zip
Loading bundle to ConductR..
[#################################################] 100%
Bundle 644a7fdbf335af40bbe1e00cce365448 is installed
Bundle loaded.
Start bundle with: conduct run 644a7fd
Unload bundle with: conduct unload 644a7fd
Print ConductR info with: conduct info
-> 0
```

**`shazar` saving with traditional usage (output slightly modified to be more scriptable)**
```bash
$ shazar conductr-crash-v0
./conductr-crash-v0-854298c953044af5ad750edb6fa76d53f612d7cd0c4b83cf11d36959bacf5e55.zip
-> 0
```

***Saving with `shazar` with traditional usage but using `-o` flag, loading with `conduct load`***
```bash
$ shazar conductr-crash-v0 -o i-can-name-my-own-files.zip
-> 0
$ conduct load i-can-name-my-own-files.zip
~/work/conductr-crash/target/bundle $ conduct load i-can-name-my-own-files.zip 
Retrieving bundle..
Retrieving file:///home/longshorej/work/conductr-crash/target/bundle/i-can-name-my-own-files.zip
Loading bundle to ConductR..
[#################################################] 100%
Bundle 854298c953044af5ad750edb6fa76d53 is installed
Bundle loaded.
Start bundle with: conduct run 854298c
Unload bundle with: conduct unload 854298c
Print ConductR info with: conduct info
-> 0
```

**Loading old format**
```bash
$ conduct load visualizer
Retrieving bundle..
Loading bundle from cache typesafe/bundle/visualizer
Bintray credentials loaded from /home/longshorej/.lightbend/commercial.credentials
Retrieving from cache /home/longshorej/.conductr/cache/bundle/visualizer-v2-73595ecbdad1f01a05db5304046b4ad5c0a98ef83e201e507a153cea21087078.zip
Loading bundle to ConductR..
[#################################################] 100%
Bundle 73595ecbdad1f01a05db5304046b4ad5 is installed
Bundle loaded.
Start bundle with: conduct run 73595ec
Unload bundle with: conduct unload 73595ec
Print ConductR info with: conduct info
-> 0
```

### CODE REVIEW

**Comment by longshorej** [comment]
[Enhance shazar](https://trello.com/c/wt4CpxMV/568-enhance-shazar)

**Review by huntc — APPROVED** [review]
Should we include a separate commit for "conduct load" so that we don't break things accidentally ie same PR.

**Comment by longshorej** [comment]
Yes, I think that's a good idea. I'll let this sit until `conductr load` is ready and then solicit more reviews.

**Comment by longshorej** [comment]
PR first comment updated

**Comment by longshorej** [comment]
[Enhance "conduct load"](https://trello.com/c/bhzQUyK7/569-enhance-conduct-load)

File: conductr_cli/resolvers/stdin_resolver.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+def resolve_bundle_version(uri):
+    return None
+
+
+def continuous_delivery_uri(resolved_version):
+    return None
+
+
+def is_bundle_name(uri):
+    return uri.count('/') == 0 and uri.count('.') == 0
```
Comment (huntc) [inline conductr_cli/resolvers/stdin_resolver.py:55]: I like how this new mechanism uses a resolver.

**Review by huntc — APPROVED** [review]

**Review by fsat — APPROVED** [review]

**Pull request merged**